### ISSUE TITLE
SDK causes crash for Instant Apps running on 6.0.1

### ISSUE THREAD

**Issue by DrewCarlson** [bugsnag/bugsnag-android#476]
### Expected behavior
InstantApps with the Bugsnag SDK should launch without a crash.

### Observed behavior
On a Nexus 5 running Android 6.0.1 an Instant App with Bugsnag enabled will crash with the following exception:
```
PID: [12022] MethodInvocationStub : Exception while calling method registerReceiver
    java.lang.SecurityException
        at java.lang.reflect.Constructor.newInstance(Native Method)
        at com.google.android.instantapps.supervisor.reflect.ReflectionUtils.a(PG:4)
        at com.google.android.instantapps.supervisor.reflect.ReflectionUtils.a(PG:33)
        at com.google.android.instantapps.supervisor.ipc.ProxyMethodHandler.invokeHandlerMethod(PG:4)
        at com.google.android.instantapps.supervisor.ipc.ProxyMethodHandler.handleMethod(PG:17)
        at com.google.android.instantapps.supervisor.ipc.base.MethodInvocationStub.onTransact(PG:19)
        at android.os.Binder.execTransact(Binder.java:453)
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at com.google.android.instantapps.supervisor.reflect.ReflectionUtils.a(PG:32)
        at com.google.android.instantapps.supervisor.ipc.ProxyMethodHandler.invokeHandlerMethod(PG:4) 
        at com.google.android.instantapps.supervisor.ipc.ProxyMethodHandler.handleMethod(PG:17) 
        at com.google.android.instantapps.supervisor.ipc.base.MethodInvocationStub.onTransact(PG:19) 
        at android.os.Binder.execTransact(Binder.java:453) 
     Caused by: java.lang.SecurityException: Broadcast android.media.RINGER_MODE_CHANGED not available to instant apps
        at com.google.android.instantapps.supervisor.ipc.common.SandboxEnforcer.enforceUnsupportedBroadcastPolicy(PG:1)
        at com.google.android.instantapps.supervisor.ipc.proxies.handler.ActivityManagerProxyHandler.registerReceiver(PG:10)
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.google.android.instantapps.supervisor.reflect.ReflectionUtils.a(PG:32) 
        at com.google.android.instantapps.supervisor.ipc.ProxyMethodHandler.invokeHandlerMethod(PG:4) 
        at com.google.android.instantapps.supervisor.ipc.ProxyMethodHandler.handleMethod(PG:17) 
        at com.google.android.instantapps.supervisor.ipc.base.MethodInvocationStub.onTransact(PG:19) 
        at android.os.Binder.execTransact(Binder.java:453)```

**Comment by fractalwrench** [comment]
Hi @DrewCarlson thanks for getting in touch. I've tried to reproduce this in an Instant App containing a single base module with Bugsnag integrated, but did not observe a crash. The behaviour I observed is that the instant app launches successfully and is able to notify bugsnag of error reports.

I tested this on an emulator and physical Nexus 5x by following [these instructions](https://developer.android.com/topic/google-play-instant/getting-started/instant-enabled-app-bundle). Were you testing in the same way or deploying via the Play Store?

**Comment by DrewCarlson** [comment]
This was discovered when launching from an internal deployment to the Play Store on a physical device.  I am also able to replicate the crash by deploying to a simulator running 6.0.0.

**Comment by fractalwrench** [comment]
@DrewCarlson I've attempted reproducing this behaviour with a modified version of [Google's sample](https://github.com/bugsnag/android-instant-apps/tree/master/aab-simple) and I'm not able to trigger a crash on a Nexus 5x x86 API 26 emulator. Does your app differ from Google's sample in any way? If you're able to reproduce this consistently it'd be really helpful to get the steps to do so, as then we'd be able to address any incompatibilities bugsnag has with Instant Apps.

**Comment by DrewCarlson** [comment]
Running your modified sample or generating an InstantApp via Android Studio and initializing the Bugsnag SDK will cause a crash on a Nexus 5 x86 v23 emulator.  Nothing special needs to be done, but I've only observed the crash on API level 23.

**Comment by fractalwrench** [comment]
Thanks for your help reproducing the issue @DrewCarlson - it seems that this occurs on physical devices running API 21 also. We're working on a fix.

**Comment by fractalwrench** [comment]
We've released a fix for this in v4.14.1, which should resolve the issue when using the `bugsnag-android` artefact.
